Map sigcontext registers for OpenBSD.  Register choice follows the Linux
block of each function: riscv64 x9/s1 (wasm instance) and x14/a4 (LLInt PC),
arm r11/r10/r8.  The armv7 mapping is untested (BROKEN-arm).

Index: Source/JavaScriptCore/runtime/MachineContext.h
--- Source/JavaScriptCore/runtime/MachineContext.h.orig
+++ Source/JavaScriptCore/runtime/MachineContext.h
@@ -187,6 +187,22 @@ static inline void*& stackPointerImpl(mcontext_t& mach
 #error Unknown Architecture
 #endif
 
+#elif OS(OPENBSD)
+
+#if CPU(X86)
+    return reinterpret_cast<void*&>((uintptr_t&) machineContext.sc_esp);
+#elif CPU(X86_64)
+    return reinterpret_cast<void*&>((uintptr_t&) machineContext.sc_rsp);
+#elif CPU(ARM)
+    return reinterpret_cast<void*&>((uintptr_t&) machineContext.sc_usr_sp);
+#elif CPU(ARM64)
+    return reinterpret_cast<void*&>((uintptr_t&) machineContext.sc_sp);
+#elif CPU(RISCV64)
+    return reinterpret_cast<void*&>((uintptr_t&) machineContext.sc_sp);
+#else
+#error Unknown Architecture
+#endif
+
 #elif OS(FUCHSIA) || OS(LINUX) || OS(HURD)
 
 #if CPU(X86_64)
@@ -304,6 +320,22 @@ static inline void*& framePointerImpl(mcontext_t& mach
 #error Unknown Architecture
 #endif
 
+#elif OS(OPENBSD)
+
+#if CPU(X86)
+    return reinterpret_cast<void*&>((uintptr_t&) machineContext.sc_ebp);
+#elif CPU(X86_64)
+    return reinterpret_cast<void*&>((uintptr_t&) machineContext.sc_rbp);
+#elif CPU(ARM)
+    return reinterpret_cast<void*&>((uintptr_t&) machineContext.sc_r11);
+#elif CPU(ARM64)
+    return reinterpret_cast<void*&>((uintptr_t&) machineContext.sc_x[29]);
+#elif CPU(RISCV64)
+    return reinterpret_cast<void*&>((uintptr_t&) machineContext.sc_s[0]);
+#else
+#error Unknown Architecture
+#endif
+
 #elif OS(FUCHSIA) || OS(LINUX) || OS(HURD)
 
 // The following sequence depends on glibc's sys/ucontext.h.
@@ -462,6 +494,22 @@ static inline void*& instructionPointerImpl(mcontext_t
 #error Unknown Architecture
 #endif
 
+#elif OS(OPENBSD)
+
+#if CPU(X86)
+    return reinterpret_cast<void*&>((uintptr_t&) machineContext.sc_eip);
+#elif CPU(X86_64)
+    return reinterpret_cast<void*&>((uintptr_t&) machineContext.sc_rip);
+#elif CPU(ARM)
+    return reinterpret_cast<void*&>((uintptr_t&) machineContext.sc_pc);
+#elif CPU(ARM64)
+    return reinterpret_cast<void*&>((uintptr_t&) machineContext.sc_elr);
+#elif CPU(RISCV64)
+    return reinterpret_cast<void*&>((uintptr_t&) machineContext.sc_sepc);
+#else
+#error Unknown Architecture
+#endif
+
 #elif OS(FUCHSIA) || OS(LINUX) || OS(HURD)
 
 // The following sequence depends on glibc's sys/ucontext.h.
@@ -634,6 +682,22 @@ inline void*& argumentPointer<1>(mcontext_t& machineCo
 #error Unknown Architecture
 #endif
 
+#elif OS(OPENBSD)
+
+#if CPU(X86)
+    return reinterpret_cast<void*&>((uintptr_t&) machineContext.sc_edx);
+#elif CPU(X86_64)
+    return reinterpret_cast<void*&>((uintptr_t&) machineContext.sc_rsi);
+#elif CPU(ARM)
+    return reinterpret_cast<void*&>((uintptr_t&) machineContext.sc_r1);
+#elif CPU(ARM64)
+    return reinterpret_cast<void*&>((uintptr_t&) machineContext.sc_x[1]);
+#elif CPU(RISCV64)
+    return reinterpret_cast<void*&>((uintptr_t&) machineContext.sc_a[1]);
+#else
+#error Unknown Architecture
+#endif
+
 #elif OS(FUCHSIA) || OS(LINUX) || OS(HURD)
 
 // The following sequence depends on glibc's sys/ucontext.h.
@@ -691,6 +755,22 @@ inline void* wasmInstancePointer(const mcontext_t& mac
 #error Unknown Architecture
 #endif
 
+#elif OS(OPENBSD)
+
+#if CPU(X86)
+    return reinterpret_cast<void*&>((uintptr_t&) machineContext.sc_esi);
+#elif CPU(X86_64)
+    return reinterpret_cast<void*&>((uintptr_t&) machineContext.sc_rbx);
+#elif CPU(ARM)
+    return reinterpret_cast<void*&>((uintptr_t&) machineContext.sc_r10);
+#elif CPU(ARM64)
+    return reinterpret_cast<void*&>((uintptr_t&) machineContext.sc_x[19]);
+#elif CPU(RISCV64)
+    return reinterpret_cast<void*&>((uintptr_t&) machineContext.sc_s[1]);
+#else
+#error Unknown Architecture
+#endif
+
 #elif OS(FUCHSIA) || OS(LINUX) || OS(HURD)
 
 // The following sequence depends on glibc's sys/ucontext.h.
@@ -804,6 +884,22 @@ inline void*& llintInstructionPointer(mcontext_t& mach
     return reinterpret_cast<void*&>((uintptr_t&) machineContext.__gregs[_REG_R8]);
 #elif CPU(ARM64)
     return reinterpret_cast<void*&>((uintptr_t&) machineContext.__gregs[_REG_X4]);
+#else
+#error Unknown Architecture
+#endif
+
+#elif OS(OPENBSD)
+
+#if CPU(X86)
+    return reinterpret_cast<void*&>((uintptr_t&) machineContext.sc_esi);
+#elif CPU(X86_64)
+    return reinterpret_cast<void*&>((uintptr_t&) machineContext.sc_r8);
+#elif CPU(ARM)
+    return reinterpret_cast<void*&>((uintptr_t&) machineContext.sc_r8);
+#elif CPU(ARM64)
+    return reinterpret_cast<void*&>((uintptr_t&) machineContext.sc_x[4]);
+#elif CPU(RISCV64)
+    return reinterpret_cast<void*&>((uintptr_t&) machineContext.sc_a[4]);
 #else
 #error Unknown Architecture
 #endif
